### DISCUSSION TITLE
Documentation: mkdocs or sphinx? I tried both and chose sphinx.

### DISCUSSION THREAD

**Discussion by HDembinski** [impy-project/chromo#110] [Ideas]
I recently added docs to pyhepmc and wanted to share some insights. I understand that Anatoli likes Markdown as much as myself, so we were wondering whether we should use mkdocs to generate the html docs, which is naturally build on Markdown and has other nice features like live-updates.

I tried to use mkdocs with the mkdocstrings plugin for pyhepmc, but I quickly ran into issues which I could not solve with those tools and so I switched back to sphinx with the autodoc extension. The autodoc plugin for sphinx works so much better than the mkdocstrings plugin for mkdocs and it offers the configuration options that I need. mkdocs and especially mkdocstrings are young projects which still lack a lot of features.

The technical setup for sphinx is very straight forward and easy. The configuration is a python file, which makes it easy to generate files dynamically for the documentation in Python. Something that mkdocs only provides via an extension. The sphinx language is RST, which is a bit more verbose than Markdown, but which also makes some things easier, like configuring drawing options for images and referencing Python classes and functions in the text.

I should say that sphinx also supports Markdown with extensions, so you can have both Markdown and autodoc. However, the aforementioned disadvantages of using Markdown for technical documentation still apply. After all, most of the documentation will be generated from the docstrings in Python, and only some framing text will be written in RST. IMHO it is not worth bothering with loading an extra extension to handle that part in Markdown.

**Comment by afedynitch** [comment]
Hmm. Thanks for providing insight into your research on this topic. It's exactly as you say, I also like MD for everything. But yaeh, there are likely some annoying technical limitations. For all other projects, I use sphinx and I also find it easy to configure and use. From my side, sphinx is totally ok!

**Comment by afedynitch** [comment]
We're now much closer to a full release and will be presenting/advertising the package everywhere soon. So, docs are becoming a more pressing issue.

Since sphinx seems the more favored option so far, I'd still opt for it. In other projects I was building and hosting it on readthedocs. Is this still a popular solution? Any reason to choose something else?
